### PR TITLE
Fix rep tests & some refactors

### PR DESCRIPTION
**Pull request by qinzuoyan**
for issue https://github.com/imzhenyu/rDSN/issues/41

### CODE REVIEW

File: src/apps/replication/client_lib/replication_common.h [hunk 1]
```diff
@@ -97,6 +97,7 @@ class replication_options
     int32_t log_file_size_mb;
     int32_t log_batch_buffer_KB_shared;
     int32_t log_batch_buffer_KB_private;
+    // TODO(qinzuoyan): the following log options is not used anywhere
```
Comment (imzhenyu) [inline src/apps/replication/client_lib/replication_common.h:100]: U r right- let's remove it

File: src/apps/replication/lib/mutation_log.cpp [hunk 2]
```diff
@@ -97,6 +97,7 @@ void mutation_log::set_valid_log_offset_before_open(global_partition_id gpid, in
     {
         _private_gpid = gpid;
         _private_valid_start_offset = valid_start_offset;
+        _private_max_decree = valid_start_offset;
```
Comment (imzhenyu) [inline src/apps/replication/lib/mutation_log.cpp]: ???

File: src/apps/replication/client_lib/replication_common.h (hunk 1 above)
Comment (qinzuoyan) [inline src/apps/replication/client_lib/replication_common.h:100]: Not only that, the log related code is some complicated, and some bugs exists, I will look into it, add some comments, fix the bugs.

File: src/apps/replication/lib/mutation_log.cpp (hunk 2 above)
Comment (qinzuoyan) [inline src/apps/replication/lib/mutation_log.cpp]: you can just ignore it

Comment (imzhenyu) [inline src/apps/replication/lib/mutation_log.cpp]: this is obviously wrong, how we can ignore it?

Comment (qinzuoyan) [inline src/apps/replication/lib/mutation_log.cpp]: I means that it's my bad and I'll remove it, so you can ignore this line.

**Comment by imzhenyu** [comment]
@qinzuoyan  it is a little bit difficult to review the code, would you mind explicitly pointing out where are the bug fixes and where else are simply code refactoring. It will save quite some time for code review. Thx.

**Comment by qinzuoyan** [comment]
In this commit, there is no bug fixed, just some refactors and some comments. But the rep_tests are fixed.

File: src/apps/replication/lib/mutation_log.cpp (hunk 2 above)
Comment (imzhenyu) [inline src/apps/replication/lib/mutation_log.cpp]: got it. np.

**Comment by imzhenyu** [comment]
I c, that's make my life much easier:-)

File: src/apps/replication/lib/mutation_log.h [hunk 3]
```diff
@@ -313,7 +324,7 @@ class log_file : public ref_counter
     std::shared_ptr<log_block> prepare_log_block() const;
 
     // async write log entry into the file
-    // 'bb' is the date to be write
+    // 'block' is the date to be writen
```
Comment (imzhenyu) [inline src/apps/replication/lib/mutation_log.h:327]: data

**Comment by imzhenyu** [comment]
seems good to me. the refactoring of replica initialization is nice.

**Comment by qinzuoyan** [comment]
ok, I am refactoring replication_options and commit logs

**Pull request merged**